File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    size = [0 for _ in range(max(tangerine) + 1)]
    for i in tangerine:
        size[i] += 1
    size = [i for i in size if i > 0]
    size.sort(reverse = True)
    sum = 0
    for i in range(len(size)):
        sum += size[i]
        if sum >= k:
            return i+1
        
    

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        Map<Integer, Integer> countMap = new HashMap<>();
        
        for(int tanger : tangerine) {
            countMap.put(tanger, countMap.getOrDefault(tanger, 0) + 1);
        }
        
        List<Integer> counts = new ArrayList<>(countMap.values());
        counts.sort((a, b) -> b - a);
        
        int cnt = 0;
        int types = 0;
        
        for (int count : counts) {
            cnt += count;
            types++;
            
            if (cnt >= k) {
                break;
            }
        }
        
        return types;
    }
}